### DISCUSSION TITLE
Can't verify provided information

### DISCUSSION THREAD

**Discussion by omprakash-prajapati** [mynaparrot/plugNmeet-server#386] [Q&A] [answered]
import hmac, hashlib, requests, json

I want to use the API and create room 

I have API_KEY, API_SECRET AND SERVER_URL

This is my code

API_URL = "https://demo.plugnmeet.cloud/auth" # dummy server url
    API_KEY = MY_API_KEY
    API_SECRET = MY_API_SECRET

    # Define the JSON body
    roomId = "room01"
    body = {
        "room_id": roomId,
    }
    json_body = json.dumps(body)

    # Generate the HMAC-SHA256 signature
    hmac_sha256 = hmac.new(
        key=bytes(API_SECRET, 'utf-8'),
        msg=bytes(json_body, 'utf-8'),
        digestmod=hashlib.sha256
    )
    signature = hmac_sha256.hexdigest()

    params = {
        "room_id": roomId,
        "metadata": {
            "room_title": "Wleness Online Another New Meeting is Here",
            "welcome_message": "Welcome to room",
            "room_features": {
                "allow_webcams": True,
                "mute_on_start": False,
                "allow_screen_share": True,
                "allow_rtmp": True,
                "admin_only_webcams": False,
                "allow_view_other_webcams": True,
                "allow_view_other_users_list": True,
                "allow_polls": True,
                "enable_analytics": True,
                "room_duration": 60,
                "recording_features": {
                    "is_allow": True,
                    "is_allow_cloud": True,
                    "is_allow_local": True,
                    "enable_auto_cloud_recording": False
                },
                "chat_features": {
                    "allow_chat": True,
                    "allow_file_upload": True
                },
                "shared_note_pad_features": {
                    "allowed_shared_note_pad": True
                },
                "whiteboard_features": {
                    "allowed_whiteboard": True
                },
                "external_media_player_features": {
                    "allowed_external_media_player": True
                },
                "waiting_room_features": {
                    "is_active": False
                },
                "breakout_room_features": {
                    "is_allow": True,
                    "allowed_number_rooms": 2
                },
                "display_external_link_features": {
                    "is_allow": True
                },
                "ingress_features": {
                    "is_allow": True
                },
                "speech_to_text_translation_features": {
                    "is_allow": True,
                    "is_allow_translation": True
                },
                "end_to_end_encryption_features": {
                    "is_enabled": False
                }
            },
            "default_lock_settings": {
                "lock_microphone": False,
                "lock_webcam": False,
                "lock_screen_sharing": True,
                "lock_whiteboard": True,
                "lock_shared_notepad": True,
                "lock_chat": False,
                "lock_chat_send_message": False,
                "lock_chat_file_share": False,
                "lock_private_chat": False
            }
        }
    }

    url = API_URL + '/room/create'
    response = requests.post(
        url,
        headers = {
            'API-KEY': API_KEY,
            'HASH-SIGNATURE': signature,
            'Content-Type': 'application/json'
        },
        params = params
    )

What could be the possible error.

**Comment by jibon57** [comment, marked as answer]
`signature` should be calculated the data you're sending. In your case `params`

**Reply by omprakash-prajapati** [reply]
It still shows the same error
